### PR TITLE
Feature/if-directive

### PR DESCRIPTION
**Pull request by kimastrom**
# Feature/if-directive

Suggestion on if directive structure

### CODE REVIEW

File: src/lib/types/directives.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+export interface ViewDirectives {
+  nIf?: boolean
+}
+
+export interface TextDirectives {
+  nIf?: boolean
```
Comment (mtnptrsn) [inline src/lib/types/directives.ts:2]: Maybe create a general interface instead? Named something like "IGeneralDirectives".

File: package.json [hunk 2]
```diff
@@ -18,7 +18,7 @@
   },
   "version": "0.0.1",
   "main": "build/index.js",
-  "repository": "git@github.com:goa-codebet/nimble.git",
+  "repository": "https://github.com/goa-codebet/nimble-utils",
```
Comment (mtnptrsn) [inline package.json]: I think we should use ssh here and not https.

File: src/lib/components/Text.tsx [hunk 3]
```diff
@@ -30,11 +32,20 @@ const Text: React.FunctionComponent<ITextProps & IConsumerInjectedProps> = ({
...
 
-  return (
+  const textElement = () => (
     <OText style={appliedStyle.style} {...rest}>
       {getTextChildren(rest, children as string)}
     </OText>
   )
+
+  // Go through all directives before render
+  const renderWithDirectives = () => {
```
Comment (mtnptrsn) [inline src/lib/components/Text.tsx]: I'd say you should have this logic inside the util folder, preferably in `src/lib/utils/directives.tsx`. This will make it much cleaner and less redundant. This file would look something like this:

```
const element = () => (
    <OText style={appliedStyle.style} {...rest}>
      {getTextChildren(rest, children as string)}
    </OText>
  )

return renderWithDirectives(element)
```

If you decide to do this, please don't forget to do the same inside the view component as well.

File: package.json [hunk 4]
```diff
@@ -2,9 +2,9 @@
   "name": "nimble-styling",
   "devDependencies": {
     "@types/color": "^3.0.0",
+    "@types/react": "16.7.7",
+    "@types/react-native": "0.57.4",
```
Comment (mtnptrsn) [inline package.json:6]: Were these changes required for your project to work?

File: src/lib/components/Text.tsx [hunk 5]
```diff
@@ -9,11 +9,13 @@ import getRegularStyles from '../utils/getRegularStyles'
 import { ITextProps } from '../..'
 import getTextChildren from '../utils/getTextChildren'
 import { flattenStyle } from '../utils/flattenStyle'
+import { ifDirective } from '../utils/directives'
 
 const Text: React.FunctionComponent<ITextProps & IConsumerInjectedProps> = ({
   children,
   theme,
   style,
+  nIf,
```
Comment (mtnptrsn) [inline src/lib/components/Text.tsx]: Maybe instead of destructuring directive props here you can just pass "rest" to the renderWithDirective function (when it's been refactored to the util folder (if you decide do do that)). Then you can destructure the props in that function instead. Less clutter in this file :)

**Review by mtnptrsn — CHANGES_REQUESTED** [review]
Just some thoughts on how we can structure it. Please let me know what you think.

File: package.json (hunk 2 above)
Comment (kimastrom) [reply package.json]: Totally agree, missed that change.

File: src/lib/components/Text.tsx (hunk 3 above)
Comment (kimastrom) [reply src/lib/components/Text.tsx]: Good idea 👍 I'll change it

File: package.json (hunk 4 above)
Comment (kimastrom) [reply package.json:6]: No, i don't know why package.json has changes in it. I'll checkout the package.json from develop.

File: src/lib/types/directives.ts (hunk 1 above)
Comment (kimastrom) [reply src/lib/types/directives.ts:2]: 👍

**Comment by kimastrom** [comment]
@mtnptrsn I've updated the pr now!

**Comment by mtnptrsn** [comment]
> @mtnptrsn I've updated the pr now!

Great!

File: src/lib/utils/directives.tsx [hunk 6]
```diff
@@ -0,0 +1,9 @@
+export const ifDirective = (c: any, nIf: boolean) => {
+  return nIf || typeof nIf === 'undefined' ? c : null
+}
+
+export const renderWithDirectives = (element, { nIf }: { nIf?: boolean }) => {
```
Comment (mtnptrsn) [inline src/lib/utils/directives.tsx:5]: Is this function written in this way to prepare for future directives? Otherwise this would be a bit neater:

```
return ifDirective(element, nIf)
```

**Review by mtnptrsn — CHANGES_REQUESTED** [review]

File: src/lib/utils/directives.tsx (hunk 6 above)
Comment (kimastrom) [reply src/lib/utils/directives.tsx:5]: Click will be added shortly!

**Review by mtnptrsn — APPROVED** [review]
